Dockerfile

```diff
@@ -1,8 +1,43 @@
-FROM ubuntu:22.04
-ARG RUST_TOOLCHAIN="1.72.0"
+# syntax=docker/dockerfile:1
+FROM ubuntu:22.04 as system-deps
+COPY install_system_dependencies.sh /opt/src/scripts/install_system_dependencies.sh
+RUN /opt/src/scripts/install_system_dependencies.sh
 
+FROM system-deps as rust
+ARG RUST_TOOLCHAIN="1.72.0"
 # Adding rust binaries to PATH.
 ENV PATH="$PATH:/root/.cargo/bin"
+COPY install_rust_toolchains.sh /opt/src/scripts/install_rust_toolchains.sh
+RUN /opt/src/scripts/install_rust_toolchains.sh
+
+FROM rust as cargo
+COPY --from=rust /root/.rustup /root/.rustup
+ENV PATH="$PATH:/root/.cargo/bin"
+COPY install_cargo_tools.sh /opt/src/scripts/install_cargo_tools.sh
+RUN /opt/src/scripts/install_cargo_tools.sh
+
+FROM system-deps as libgpiod
+COPY install_libgpiod.sh /opt/src/scripts/install_libgpiod.sh
+RUN /opt/src/scripts/install_libgpiod.sh
 
-COPY build_container.sh /opt/src/scripts/build_container.sh
-RUN /opt/src/scripts/build_container.sh
+# Build and install libpipewire.
+
+FROM system-deps as pipewire-deps
+COPY install_libpipewire_dependencies.sh /opt/src/scripts/install_libpipewire_dependencies.sh
+RUN /opt/src/scripts/install_libpipewire_dependencies.sh
+
+FROM pipewire-deps as libpipewire
+COPY install_libpipewire.sh /opt/src/scripts/install_libpipewire.sh
+RUN /opt/src/scripts/install_libpipewire.sh
```

Comment:
I am not sure we will be able to merge this commit. I had to reduce the number of RUN commands to one because otherwise we were out of space when trying to build the container and publishing it would fail. Some more information here: https://github.com/rust-vmm/rust-vmm-container/commit/d1a36b4b2af5487730bde630874ee92f8c87dbce and https://github.com/rust-vmm/rust-vmm-container/commit/fdd95e2d5e90868d0eca93cbc5c52cce55e82d2a

Reply:
Maybe it works now, but I am not super sure, we need to see what is the size generated by the build 😬

Reply:
@andreeaflorescu yeah that's why I closed it, forgot to comment here :) Thanks!

Reply:
I'm leaving the commits/branch though, in case the storage size situation changes in the future.